drivers/usb/udc/udc_stm32.c

```diff
@@ -50,7 +50,11 @@ LOG_MODULE_REGISTER(udc_stm32, CONFIG_UDC_DRIVER_LOG_LEVEL);
 #define HIGH_SPEED             USB_OTG_SPEED_HIGH
 #endif
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_usb)
+#define FULL_SPEED             PCD_SPEED_FULL
+#else
 #define FULL_SPEED             USB_OTG_SPEED_FULL
+#endif
```

Comment:
HIGH_SPEED/FULL_SPEED is a very bad choice. I see this was introduced in https://github.com/zephyrproject-rtos/zephyr/commit/01ddccdf626a2d767d4b76b648fc9e520c35ded5. No idea how this could get enough approvals and be merged. Not that there are no macros that are already prefixed correctly with `UDC_STM32_`.

Reply:
@jfischer-no so you are suggesting these should be renamed to `UDC_STM32_FULL_SPEED` and `UDC_STM32_FULL_SPEED`? Not super clear to me from your comment, trying to land this to an approvable state with few turnarounds since it break the build for load of platforms.

Reply:
Yes.

> it break the build for load of platforms.

Just revert https://github.com/zephyrproject-rtos/zephyr/pull/86922

Reply:
Thanks, I think it's fine to wait for a fix forward no need to rush a revert, just wanted to go straight to the right fix.

Reply:
> Just revert #86922

Given that there's still no progress on this, I suggest we perform a revert. This is breaking downstream users/CI.

Reply:
Outdated, should be fixed by https://github.com/zephyrproject-rtos/zephyr/pull/88244